hibernate-core/src/main/java/org/hibernate/metamodel/model/convert/internal/NamedEnumValueConverter.java

```diff
@@ -56,42 +55,19 @@ public int getJdbcTypeCode() {
 
 	@Override
 	public E readValue(ResultSet resultSet, String name) throws SQLException {
-		final String value = resultSet.getString( name );
-
-		final boolean traceEnabled = log.isTraceEnabled();
-		if ( resultSet.wasNull() ) {
-			if ( traceEnabled ) {
-				log.trace( String.format( "Returning null as column [%s]", name ) );
-			}
-			return null;
-		}
-
-		final E enumValue = toDomainValue( value );
-		if ( traceEnabled ) {
-			log.trace( String.format( "Returning [%s] as column [%s]", enumValue, name ) );
-		}
-
-		return enumValue;
+		return VarcharTypeDescriptor.INSTANCE.getExtractor( enumJavaDescriptor ).extract( resultSet, name, null );
```

Comment:
Shouldn't we initialize `VarcharTypeDescriptor.INSTANCE.getExtractor( enumJavaDescriptor )` once and for all in the constructor?

Same question for the binder (and obviously the Integer version).

AFAICS, the objects are stateless.

Reply:
Good point. We could definitely do that.